File: widget/src/main/java/app/allever/android/lib/widget/ClickListener.java
package app.allever.android.lib.widget;

import android.view.View;

import app.allever.android.lib.core.ext.LoggerKt;


/**
 * @author allever
 */
public abstract class ClickListener implements View.OnClickListener {

    private static final long TIME_INTERVAL = 500;

    private long lastTime;

    /**
     * @param v
     */
    public abstract void click(View v);

    @Override
    public void onClick(View v) {
        if (checkTime()) {
            LoggerKt.debugD("防止多次点击");
            return;
        }
        this.click(v);
    }

    private boolean checkTime() {
        boolean flag = true;
        long time = System.currentTimeMillis() - lastTime;

        if (time > getTimeInterval()) {
            flag = false;
        }
        lastTime = System.currentTimeMillis();
        return flag;
    }

    /**
     * 设置时间间隔， 默认1秒
     *
     * @return
     */
    protected long getTimeInterval() {
        return TIME_INTERVAL;
    }

}

File: widget/src/main/java/app/allever/android/lib/widget/ClickListener.kt
package app.allever.android.lib.widget

import android.view.View
import app.allever.android.lib.core.ext.debugD

/**
 * @author allever
 */
abstract class ClickListener : View.OnClickListener {
    private var lastTime: Long = 0

    /**
     * @param v
     */
    abstract fun click(v: View?)

    override fun onClick(v: View) {
        if (checkTime()) {
            debugD("防止多次点击")
            return
        }
        click(v)
    }

    private fun checkTime(): Boolean {
        var flag = true
        val time = System.currentTimeMillis() - lastTime
        if (time > timeInterval()) {
            flag = false
        }
        lastTime = System.currentTimeMillis()
        return flag
    }

    protected fun timeInterval(): Long {
        return TIME_INTERNAL
    }

    companion object {
        /**
         * 设置时间间隔， 默认1秒
         *
         * @return
         */
        protected const val TIME_INTERNAL: Long = 500L
    }
}